oak_functions_service/proto/oak_functions.proto

```diff
@@ -63,35 +84,36 @@ message LookupDataChunk {
   repeated LookupDataEntry items = 1;
 }
 
-// Indicates to the LookupDataManager which action should be taken.
-enum UpdateAction {
-  // The sender extends the lookup data builder with this chunk and will send a
-  // further chunk.
-  EXTEND = 0;
-  // The final chunk the sender sends to build the lookup data.
-  FINISH = 1;
-  // The sender needs to abort the building the lookup data.
-  ABORT = 2;
+// If the definition of ExtendNextLookupData changes, the estimation of the size when
+// serialized in the Oak Functions Launcher needs to change, too.
+message ExtendNextLookupDataRequest {
+  LookupDataChunk chunk = 1;
 }
 
-// A message to the LookupDataManager to process the lookup data chunk based on the current
-// action.
-//
-// If the definition of UpdateLookupDataRequest changes, the estimation of the size when
+message ExtendNextLookupDataResponse {
+  bool ok = 1;
+}
+
+// If the definition of FinishNextLookupData changes, the estimation of the size when
 // serialized in the Oak Functions Launcher needs to change, too.
-message UpdateLookupDataRequest {
-  UpdateAction action = 1;
-  LookupDataChunk chunk = 2;
+message FinishNextLookupDataRequest {
+  LookupDataChunk chunk = 1;
```

Comment:
Is this the final form of the API? I don't think we should keep the distinction between "here is a chunk, and it is not the last one" vs "here is a chunk, and it is the last one"; rather, I believe we should have "here is a chunk", and "finished, no more chunks" as separate methods.

Reply:
I have already two seperate methods, really.

Or, would you want your "finished, no more chunks" have no payload (chunk)? Then every update is at least two messages. What do you think would be the benefit?

Alternatively we could have a bool in the request which indicates `finished`. Then we'd only have one method.

The current proposal sends the least messages and data, I believe.

Reply:
The goal should be to maximize readability / maintainability. I don't think it's worth combining two semantically distinct operations (here is a chunk + done) to save one message; especially when there is already another method with the semantics of one such operations (here is a chunk). We should treat these as methods on a struct, with orthogonal types and semantics as much as possible.

> Or, would you want your "finished, no more chunks" have no payload (chunk)? Then every update is at least two messages. What do you think would be the benefit?

The benefit is readability and separation of concerns. Yes at the expense of one extra invocation, but that's an easy trade off here IMO, especially when it's adding a couple of ms to an operation that may take up to minutes / hours.

> Alternatively we could have a bool in the request which indicates finished. Then we'd only have one method.

Yes this may also work, but it still seems a premature optimization to combine two intentions in one method IMO, when we can easily have separate methods. Especially as in this case the pattern is like a builder: add entries, then declare when it's finished. A lot of types already work that way, so it's an expected pattern for developers.